Typings broken after 4.2.0

After v4.2.0 with the "big" typings change, basically everything broke. Was this even tested?

```
$ rimraf dist && tsc
../../node_modules/winston-daily-rotate-file/index.d.ts:1:8 - error TS1259: Module '"/home/runner/work/***/***/node_modules/winston-transport/index"' can only be default-imported using the 'esModuleInterop' flag

1 import TransportStream from "winston-transport";
         ~~~~~~~~~~~~~~~

  ../../node_modules/winston-transport/index.d.ts:37:1
    37 export = TransportStream;
       ~~~~~~~~~~~~~~~~~~~~~~~~~
    This module is declared with using 'export =', and can only be used with a default import when using the 'esModuleInterop' flag.

util/logger.ts:55:7 - error TS2351: This expression is not constructable.
  Type 'typeof import("/home/runner/work/***/***/node_modules/winston-daily-rotate-file/index")' has no construct signatures.

55  	new DailyRotateFile({
    	    ~~~~~~~~~~~~~~~
```

Reading this should be pretty evident whats happening here. So lets go over this, first one? I can't really do anything against, the import is simply wrong because `winston-transport` does not export a default.
https://github.com/winstonjs/winston-daily-rotate-file/blob/21fa56519bf64fbde90baa6ec74c4020b0d08a00/index.d.ts#L1

https://github.com/winstonjs/winston-transport/blob/46db8f3c8cd8b106ade8d7e04a191ee388683d60/index.d.ts#L37

Not everyone uses esModuleInterop so relying on this is quite bad.

Now what about the 2nd one. Not being able to construct DailyRotateFile? But I imported it just fine with `import * as DailyRotateFile from 'winston-daily-rotate-file';` before 4.2.0

Well this is actually incorrect in the typings, they say:
https://github.com/winstonjs/winston-daily-rotate-file/blob/21fa56519bf64fbde90baa6ec74c4020b0d08a00/index.d.ts#L106

But in reality:
https://github.com/winstonjs/winston-daily-rotate-file/blob/21fa56519bf64fbde90baa6ec74c4020b0d08a00/index.js#L6-L7

And even after fixing all of those, I can't go back to the intended way of calling `import * as DailyRotatefile from 'x';` because:
`This module can only be referenced with ECMAScript imports/exports by turning on the 'esModuleInterop' flag and referencing its default export.`

Mainly because of how the typings file is written now.

--- a/index.d.ts
+++ b/index.d.ts
@@ -1,4 +1,4 @@
-import TransportStream from "winston-transport";
+import TransportStream = require("winston-transport");
 
 // referenced from https://stackoverflow.com/questions/40510611/typescript-interface-require-one-of-two-properties-to-exist
 type RequireOnlyOne<T, Keys extends keyof T = keyof T> =
@@ -12,95 +12,97 @@ type RequireOnlyOne<T, Keys extends keyof T = keyof T> =
 // merging into winston.transports
 declare module 'winston/lib/winston/transports' {
     interface Transports {
-        DailyRotateFileTransport: typeof DailyRotateFileTransport;
-        DailyRotateFileTransportOptions: DailyRotateFileTransportOptions;
+        DailyRotateFileTransport: typeof DailyRotateFile;
+        DailyRotateFileTransportOptions: DailyRotateFile.DailyRotateFileTransportOptions;
     }
 }
 
-export type DailyRotateFileTransportOptions = RequireOnlyOne<GeneralDailyRotateFileTransportOptions, 'filename' | 'stream'>;
+declare namespace DailyRotateFile {
+    type DailyRotateFileTransportOptions = RequireOnlyOne<GeneralDailyRotateFileTransportOptions, 'filename' | 'stream'>;
 
-interface GeneralDailyRotateFileTransportOptions extends TransportStream.TransportStreamOptions {
-    json?: boolean;
-    eol?: string;
+    interface GeneralDailyRotateFileTransportOptions extends TransportStream.TransportStreamOptions {
+        json?: boolean;
+        eol?: string;
 
-    /**
-     * A string representing the moment.js date format to be used for rotating. The meta characters used in this string will dictate the frequency of the file rotation. For example, if your datePattern is simply 'HH' you will end up with 24 log files that are picked up and appended to every day. (default 'YYYY-MM-DD')
-     */
-    datePattern?: string;
+        /**
+         * A string representing the moment.js date format to be used for rotating. The meta characters used in this string will dictate the frequency of the file rotation. For example, if your datePattern is simply 'HH' you will end up with 24 log files that are picked up and appended to every day. (default 'YYYY-MM-DD')
+         */
+        datePattern?: string;
 
-    /**
-     * A boolean to define whether or not to gzip archived log files. (default 'false')
-     */
-    zippedArchive?: boolean;
+        /**
+         * A boolean to define whether or not to gzip archived log files. (default 'false')
+         */
+        zippedArchive?: boolean;
 
-    /**
-     * Filename to be used to log to. This filename can include the %DATE% placeholder which will include the formatted datePattern at that point in the filename. (default: 'winston.log.%DATE%)
-     */
-    filename?: string;
+        /**
+         * Filename to be used to log to. This filename can include the %DATE% placeholder which will include the formatted datePattern at that point in the filename. (default: 'winston.log.%DATE%)
+         */
+        filename?: string;
 
-    /**
-     * The directory name to save log files to. (default: '.')
-     */
-    dirname?: string;
+        /**
+         * The directory name to save log files to. (default: '.')
+         */
+        dirname?: string;
 
-    /**
-     * Write directly to a custom stream and bypass the rotation capabilities. (default: null)
-     */
-    stream?: NodeJS.WritableStream;
+        /**
+         * Write directly to a custom stream and bypass the rotation capabilities. (default: null)
+         */
+        stream?: NodeJS.WritableStream;
 
-    /**
-     * Maximum size of the file after which it will rotate. This can be a number of bytes, or units of kb, mb, and gb. If using the units, add 'k', 'm', or 'g' as the suffix. The units need to directly follow the number. (default: null)
-     */
-    maxSize?: string | number;
+        /**
+         * Maximum size of the file after which it will rotate. This can be a number of bytes, or units of kb, mb, and gb. If using the units, add 'k', 'm', or 'g' as the suffix. The units need to directly follow the number. (default: null)
+         */
+        maxSize?: string | number;
 
-    /**
-     * Maximum number of logs to keep. If not set, no logs will be removed. This can be a number of files or number of days. If using days, add 'd' as the suffix. (default: null)
-     */
-    maxFiles?: string | number;
+        /**
+         * Maximum number of logs to keep. If not set, no logs will be removed. This can be a number of files or number of days. If using days, add 'd' as the suffix. (default: null)
+         */
+        maxFiles?: string | number;
 
-    /**
-     * An object resembling https://nodejs.org/api/fs.html#fs_fs_createwritestream_path_options indicating additional options that should be passed to the file stream. (default: `{ flags: 'a' }`)
-     */
-    options?: string | object;
+        /**
+         * An object resembling https://nodejs.org/api/fs.html#fs_fs_createwritestream_path_options indicating additional options that should be passed to the file stream. (default: `{ flags: 'a' }`)
+         */
+        options?: string | object;
 
-    /**
-     * A string representing the name of the name of the audit file. (default: './hash-audit.json')
-     */
-    auditFile?: string;
+        /**
+         * A string representing the name of the name of the audit file. (default: './hash-audit.json')
+         */
+        auditFile?: string;
 
-    /**
-     * A string representing the frequency of rotation. (default: 'custom')
-     */
-    frequency?: string;
+        /**
+         * A string representing the frequency of rotation. (default: 'custom')
+         */
+        frequency?: string;
 
-    /**
-     * A boolean whether or not to generate file name from "datePattern" in UTC format. (default: false)
-     */
-    utc?: boolean;
+        /**
+         * A boolean whether or not to generate file name from "datePattern" in UTC format. (default: false)
+         */
+        utc?: boolean;
 
-    /**
-     * A string representing an extension to be added to the filename, if not included in the filename property. (default: '')
-     */
-    extension?: string;
+        /**
+         * A string representing an extension to be added to the filename, if not included in the filename property. (default: '')
+         */
+        extension?: string;
 
-    /**
-      * Create a tailable symlink to the current active log file. (default: false)
-      */
-    createSymlink?: boolean;
+        /**
+         * Create a tailable symlink to the current active log file. (default: false)
+         */
+        createSymlink?: boolean;
 
-    /**
-      * The name of the tailable symlink. (default: 'current.log')
-      */
-    symlinkName?: string;
+        /**
+         * The name of the tailable symlink. (default: 'current.log')
+         */
+        symlinkName?: string;
+    }
 }
 
-declare class DailyRotateFileTransport extends TransportStream {
+declare class DailyRotateFile extends TransportStream {
     filename: string;
     dirname: string;
     logStream: NodeJS.WritableStream;
-    options: DailyRotateFileTransportOptions;
+    options: DailyRotateFile.DailyRotateFileTransportOptions;
 
-    constructor(options: DailyRotateFileTransportOptions);
+    constructor(options?: DailyRotateFile.DailyRotateFileTransportOptions);
 }
 
-export default DailyRotateFileTransport;
+export = DailyRotateFile;

